docs/security.html

```diff
@@ -36,7 +36,104 @@ <h3 class="anchor-heading"><a id="security_overview" class="anchor-link"></a><a
+      by the security protocol defined by <code>security.inter.broker.protocol</code>, which
+      defaults to <code>PLAINTEXT</code>.</p>
+    
+    <p>For legacy clusters which rely on Zookeeper to store cluster metadata, it is possible to
+      declare a separate listener to be used for metadata propagation from the active controller
+      to the brokers. This is defined by <code>control.plane.listener.name</code>. The active controller
+      will use this listener when it needs to push metadata updates to the brokers in the cluster.
+      The benefit of using a control plane listener is that it uses a separate processing thread,
+      which makes it less likely for application traffic to impede timely propagation of metadata changes
+      (such as partition leader and ISR updates). Note that the default value is null, which
+      means that the controller will use the same listener as defined by <code>inter.broker.listener</code></p>
+    
+    <p>In a KRaft cluster, the listener defined by <code>inter.broker.listener.name</code> is used
+      exclusively for requests sent to nodes which have the "broker" role enabled in <code>process.roles</code>.
+      For legacy clusters, the active controller will connect to all brokers in the cluster
+      to send metadata updates. In KRaft, it is the other way around. The broker is responsible
+      for finding the current controller and connecting to it in order to receive metadata updates.
+      Hence the configuration <code>control.plane.listener.name</code> cannot be used in KRaft clusters.</p>
+      
+    <p>Instead, nodes with the "controller" role must define separate listeners through the
+      <code>controller.listener.names</code> configuration. Any node running with both the
+      "broker" and "controller" roles must use separate listeners for the broker and controller.
+      For example, a minimal configuration for a combined node might look like the following:</p>
+    
+    <pre class="line-numbers"><code class="language-text">process.roles=broker,controller
+listeners=BROKER://localhost:9092,CONTROLLER://localhost:9093
+inter.broker.listener.name=BROKER
+controller.listener.names=CONTROLLER
+listener.security.protocol.map=BROKER:SASL_SSL,CONTROLLER:SASL_SSL</code></pre>
+
+    <p>This defines two listeners, one for the controller and one for the broker. It is conventional
+      to use a separate listener for client traffic as well.</p>
+    
+    <p>The controller will accept requests on all listeners defined by <code>controller.listener.names</code>.
+      For outbound requests to other controller nodes, the first listener listed in <code>controller.listener.names</code>
+      will be used. The list of listeners provides a way to change the active listener from one port
+      to another through a roll of the cluster. Basically do one roll to expose the new listener,
+      and one roll to remove the old listener.</p>
```

Comment:
```suggestion
    <p>The controller will accept requests on all listeners defined by <code>controller.listener.names</code>.
      For outbound requests to other controller servers, the first listener listed in <code>controller.listener.names</code>
      will be used. The list of listeners provides a way to change the active listener from one port
      to another through a roll of the cluster. Basically do one roll to expose the new listener,
      and one roll to remove the old listener.</p>
```

Hmm. Regarding the last sentence, maybe we can add a section at the end describing the process of changing the controller listener. What do you think? I think that last sentence is hiding a lot of details.